core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -815,3 +817,24 @@ class DynamicListenerConfig(server: KafkaServer) extends BrokerReconfigurable wi
 
 }
 
+object DynamicConnectionQuota {
+  val ReconfigurableConfigs = Set(KafkaConfig.MaxConnectionsPerIpProp, KafkaConfig.MaxConnectionsPerIpOverridesProp)
+}
+
+class DynamicConnectionQuota(server: KafkaServer) extends BrokerReconfigurable {
+
+  override def reconfigurableConfigs: Set[String] = {
+    DynamicConnectionQuota.ReconfigurableConfigs
+  }
+
+  override def validateReconfiguration(newConfig: KafkaConfig): Unit = {
```

Comment:
Maybe this would be a good time to check for valid IP addresses?

Also, there is a requirement in `KafkaConfig` that the default value can only be set to 0 if there are per-ip overrides. Should we check that?

Reply:
Above requirement will validated be during newConfig creation via kafkaConfig.validateValues. 
newly added DynamicBrokerConfigTest.testConnectionQuota test validates this scenario.

We can add check for valid valid IP addresses.  Would it be OK, If I use Apache commons validator library? This will give Ipv4, ipv6 validation util class.
http://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/InetAddressValidator.html

Reply:
@hachikuji Can you take one more look? Also above query?

Reply:
I wasn't imagining anything too elaborate. Just perhaps using something like `ClientUtils.parseAndValidateAddresses`.

Reply:
@hachikuji I have added basic validation for host address.  not sure, if this meets your expectation.  Please take a look.

Reply:
@hachikuji Can you please take look?